pkgs/by-name/id/idb-companion/package.nix

```
@@ -0,0 +1,36 @@
+{ lib
+, stdenv
+, fetchurl
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "idb-companion";
+  version = "1.1.8";
+
+  src = fetchurl {
+    url = "https://github.com/facebook/idb/releases/download/v${finalAttrs.version}/idb-companion.universal.tar.gz";
+    hash = "sha256-O3LMappbGiKhiCBahAkNOilDR6hGGA79dVzxo8hI4+c=";
+  };
+
+  sourceRoot = "idb-companion.universal";
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin
+    cp -r ./* $out/bin/
+    ln -s $out/bin/bin/idb_companion $out/bin/idb_companion
+
+    runHook postInstall
+  '';
```

I'm reading what you're doing there...

```suggestion
  installPhase = ''
    runHook preInstall

    mkdir -p $out
    cp -r bin $out/

    runHook postInstall
  '';
```

How about just doing this? ^^

---

@drupol : The problem with this approach is that the Framework folder which is necessary gets left out:

```
[nix-shell:~/code/experiments/status-mobile]$ idb_companion
dyld[51661]: Library not loaded: @rpath/FBControlCore.framework/Versions/A/FBControlCore
  Referenced from: <86D0A585-D0B1-30C8-BA25-9F3E3AB1E2DB> /nix/store/l7p3bs1fgr36597jj65qg3n8py3mzbdv-idb-companion-1.1.8/bin/idb_companion
  Reason: tried: '/usr/lib/swift/FBControlCore.framework/Versions/A/FBControlCore' (no such file, not in dyld cache), '/System/Volumes/Preboot/Cryptexes/OS/usr/lib/swift/FBControlCore.framework/Versions/A/FBControlCore' (no such file), '/nix/store/l7p3bs1fgr36597jj65qg3n8py3mzbdv-idb-companion-1.1.8/Frameworks/FBControlCore.framework/Versions/A/FBControlCore' (no such file), '/usr/lib/swift/FBControlCore.framework/Versions/A/FBControlCore' (no such file, not in dyld cache), '/System/Volumes/Preboot/Cryptexes/OS/usr/lib/swift/FBControlCore.framework/Versions/A/FBControlCore' (no such file), '/nix/store/l7p3bs1fgr36597jj65qg3n8py3mzbdv-idb-companion-1.1.8/Frameworks/FBControlCore.framework/Versions/A/FBControlCore' (no such file), '/Library/Frameworks/FBControlCore.framework/Versions/A/FBControlCore' (no such file), '/System/Library/Frameworks/FBControlCore.framework/Versions/A/FBControlCore' (no such file, not in dyld cache)
Abort trap: 6
```

---

And just doing `cp -r . $out/` ?

---

that works, thanks for helping in simplifying this 👍🏻